src/options.rs

```diff
@@ -0,0 +1,284 @@
+                self.parse_setters_options_metaItem(meta_item)
+            } else {
+                panic!("Expected NestedMetaItem::MetaItem, found {:?}", x)
+            }
+        }
+    }
+
+    #[allow(non_snake_case)]
+    fn parse_setters_options_metaItem(&mut self, meta_item: &syn::MetaItem) {
+        trace!("Parsing MetaItem {:?}", meta_item);
+        match *meta_item {
+            syn::MetaItem::Word(ref ident) => {
+                self.parse_setters_options_word(ident)
+            },
+            syn::MetaItem::NameValue(ref ident, ref lit) => {
+                self.parse_setters_options_nameValue(ident, lit)
+            },
+            _ => {
+                panic!("Expected MetaItem::Word/NameValue, found {:?}", meta_item)
+            }
+        }
+    }
+
+    /// e.g `owned` in `#[setters(owned)]`
+    fn parse_setters_options_word(&mut self, ident: &syn::Ident) {
+        trace!("Parsing word {:?}", ident);
+        match ident.as_ref() {
+            "owned" => {
+                self.setter_pattern(SetterPattern::Owned)
+            },
+            "mutable" => {
+                self.setter_pattern(SetterPattern::Mutable)
+            },
+            "immutable" => {
+                self.setter_pattern(SetterPattern::Immutable)
+            },
+            "public" => {
+                self.setter_public(true)
+            },
+            "private" => {
+                self.setter_public(false)
+            },
+            _ => {
+                panic!("Unknown option {:?}", ident)
+            }
+        };
+    }
+
+    /// e.g `prefix="with"` in `#[setters(prefix="with")]`
+    #[allow(non_snake_case)]
+    fn parse_setters_options_nameValue(&mut self, ident: &syn::Ident, lit: &syn::Lit) {
+        trace!("Parsing named value {:?} = {:?}", ident, lit);
+        match ident.as_ref() {
+            "prefix" => {
+                self.parse_setters_prefix(lit)
+            },
+            "option" => {
+                self.parse_setters_implicit_options(lit)
+            }
+            _ => {
+                panic!("Unknown option {:?}", ident)
+            }
+        }
+    }
+
+    fn parse_setters_prefix(&mut self, lit: &syn::Lit) {
+        trace!("Parsing prefix {:?}", lit);
+        let value = parse_lit_as_cooked_string(lit);
+        debug!("Setting prefix {:?}", value);
+        self.setter_prefix = Some(value.clone());
+    }
+
+    fn parse_setters_implicit_options(&mut self, lit: &syn::Lit) {
+        trace!("Parsing implicit options {:?}", lit);
+        let value = parse_lit_as_cooked_string(lit);
+        match value.as_str() {
+            "implicit" => self.setter_implicit_options(true),
+            "explicit" => self.setter_implicit_options(true),
+            _ => panic!("Unknown value {:?}", value),
+        };
+    }
+
+    fn parse_getters_options(&mut self, nested: &[syn::NestedMetaItem]) -> &mut Self {
+        panic!("TODO: parse getter options -> {:?}", nested);
```

Comment:
Actually in published code I would like to get rid of these unimplemented functions. But generally you are right. :-)